keel-core/src/main/kotlin/com/netflix/spinnaker/keel/persistence/ArtifactRepository.kt

```diff
@@ -208,6 +220,46 @@ interface ArtifactRepository : PeriodicallyCheckedRepository<DeliveryArtifact> {
    * different values.
    */
   override fun itemsDueForCheck(minTimeSinceLastCheck: Duration, limit: Int): Collection<DeliveryArtifact>
+
+  /**
+   * Returns true if the new version is lower than the existing version.
+   * Note: the artifact comparitors are decending by default
+   */
+  fun isLower(artifact: DeliveryArtifact, new: String, existing: String): Boolean =
+    artifact.versioningStrategy.comparator.compare(new, existing) > 0
+
+  /**
+   * Returns true if the new version is higher than the existing version.
+   * Note: the artifact comparitors are decending by default
+   */
+  fun isHigher(artifact: DeliveryArtifact, new: String, existing: String): Boolean =
+    artifact.versioningStrategy.comparator.compare(new, existing) < 0
+
+  /**
```

Comment:
General question - we are relaying here on the fact that a newer version will get the higher number. is it always the case?

Reply:
Yes! It is always the case that we are deciding which is the newer version by this comparator, and "higher" version will always be newer. It's one of our core artifact assumptions. It's definitely an assumption, but it's what allows us to do cool artifact things.